noty-android/core/src/main/java/dev/shreyaspatil/noty/core/connectivity/ConnectivityObserver.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2020 Shreyas Patil
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dev.shreyaspatil.noty.core.connectivity
+
+import kotlinx.coroutines.flow.Flow
+
+/**
+ * A utility for observing the connectivity status
+ */
+interface ConnectivityObserver {
+    /**
+     * Gives the realtime updates of a [ConnectionState]
+     */
+    val connectionState: Flow<ConnectionState>
+
+    /**
+     * Retrieves the current [ConnectionState]
+     */
+    val currentConnectionState: ConnectionState
```

Comment:
Instead of maintaining two fields for a current state and having a stream of state, can we have a `StateFlow<ConnectionState>` which can suffice both the use cases?